src/file.bash

```diff
@@ -332,24 +332,23 @@ assert_file_owner() {
 assert_file_permission() {
   local -r permission="$1"
   local -r file="$2"
-  if [[ `uname` == "Darwin" ]]; then
-  if [ `stat -f '%A' "$file"` -ne "$permission" ]; then
-    local -r rem="${BATSLIB_FILE_PATH_REM-}"
-    local -r add="${BATSLIB_FILE_PATH_ADD-}"
-    batslib_print_kv_single 4 'path' "${file/$rem/$add}" \
-      | batslib_decorate "file does not have permissions $permission" \
-      | fail
-  fi
-  elif [[ `uname` == "Linux" ]]; then
-  if [ `stat -c "%a" "$file"` -ne "$permission" ]; then
+
+  case "$OSTYPE" in
+    darwin*)
+      local -r actual_permission=$(stat -f '%A' "$file")
+    ;;
+    linux*)
```

Comment:
What is `uname`'s value on MSYS?

Reply:
Decided to check only for MacOS and regard the rest as linux. The code does not seem to be written with MSYS, Cygwin (or WSL) in mind.